### PR TITLE
fix issue1057: add subsetting label functionality

### PR DESCRIPTION
**Pull request by smilesun**
1. i changed partly the old code, as there is a sub-bug, for example 

> names(surv.df)
>  [1] "time"   "status" "x1"     "x2"     "x3"     "x4"     "x5"     "x6"     "x7"     "x8"     "x9"   

after subseting the data, the old code will turn it into something 

[1]   "x6"     "x7"     "x8"     "x9"   "time"   "status"   
1. i can't resolve the error locally regarding the different version of java which is required to install 'RWeka', so i just use travis to run the tests like '''test('~/mlr', filter = 'base_measures')'''

### CODE REVIEW

File: R/Task_operators.R [hunk 1]
```diff
@@ -279,6 +280,29 @@ getTaskData = function(task, subset, features, target.extra = FALSE, recode.targ
   }
 
   tn = task$task.desc$target
+  ############
+  extractLabel = function(tn, labels){
+      assert(checkIntegerish(labels, lower = 1L, upper = length(tn)), checkLogical(labels), checkCharacter(labels))
+      if (is.numeric(labels)){
+        labels = asInteger(labels)
```
Comment (larskotthoff) [inline R/Task_operators.R]: Why is this conversion necessary?

File: R/Task_operators.R [hunk 2]
```diff
@@ -279,6 +280,29 @@ getTaskData = function(task, subset, features, target.extra = FALSE, recode.targ
   }
 
   tn = task$task.desc$target
+  ############
```
Comment (larskotthoff) [inline R/Task_operators.R]: Comment line like this not needed.

File: R/Task_operators.R [hunk 3]
```diff
@@ -279,6 +280,29 @@ getTaskData = function(task, subset, features, target.extra = FALSE, recode.targ
...
+  extractLabel = function(tn, labels){
+      assert(checkIntegerish(labels, lower = 1L, upper = length(tn)), checkLogical(labels), checkCharacter(labels))
+      if (is.numeric(labels)){
```
Comment (larskotthoff) [inline R/Task_operators.R]: Style guide: indent 2 spaces.

Comment (larskotthoff) [inline R/Task_operators.R]: Style guide: space before {

File: R/Task_operators.R [hunk 4]
```diff
@@ -279,6 +280,29 @@ getTaskData = function(task, subset, features, target.extra = FALSE, recode.targ
...
+        labels = asInteger(labels)
+      }
+      if (!is.character(labels)){
+        rlabels = tn[labels] # if labels are logical not enought length, the remaining will be default be be TRUE
```
Comment (larskotthoff) [inline R/Task_operators.R]: You're assuming that the labels are logical here; why not check for that?

File: R/Task_operators.R [hunk 5]
```diff
@@ -279,6 +280,29 @@ getTaskData = function(task, subset, features, target.extra = FALSE, recode.targ
...
+      }
+      else {
```
Comment (larskotthoff) [inline R/Task_operators.R]: Style guide!

File: tests/testthat/test_base_getTaskData.R [hunk 6]
```diff
@@ -85,3 +86,8 @@ test_that("getTaskData multilabel", {
   x = getTaskData(multilabel.task, target.extra = TRUE)
   expect_equal(dim(x$target), c(150L, 2L))
 })
+
+test_that("getTaskData yeast_multilabel", {
+  df_yeast = getTaskData(yeast.task, labels = c(TRUE,FALSE))
+  expect_true(!is.null(setdiff(getTaskTargetNames(yeast.task),names(df_yeast))))
```
Comment (larskotthoff) [inline tests/testthat/test_base_getTaskData.R]: Check explicitly for what should be there and what shouldn't be there.

**Comment by larskotthoff** [comment]
What's the status here @smilesun ?

**Comment by pat-s** [comment]
I can't figure out what this PR does and we have no update in two years, can we close here?

**Comment by pat-s** [comment]
See last comment.

**Pull request closed without merging**